Fix the build on 10.4/10.5. The configure script already checks for
pthread_getname_np, so just use the result of that to decide whether
pthread_setname_np is available. (Patching the configure script itself
introduces timestamp and patch-maintenance headaches.)

--- js/src/threading/posix/Thread.cpp.orig
+++ js/src/threading/posix/Thread.cpp
@@ -152,9 +152,11 @@
   name = nameBuf;
 #endif
 
-  int rv;
+  int rv = 0;
 #ifdef XP_DARWIN
+#ifdef HAVE_PTHREAD_GETNAME_NP
   rv = pthread_setname_np(name);
+#endif
 #elif defined(__DragonFly__) || defined(__FreeBSD__) || defined(__OpenBSD__)
   pthread_set_name_np(pthread_self(), name);
   rv = 0;
